data/logs_model/splunk_logs_model.py

```diff
@@ -0,0 +1,85 @@
+import json
+import logging
+
+from datetime import datetime
+
+from data import model
+from data.logs_model.logs_producer import LogProducerProxy, LogSendException
+from data.logs_model.logs_producer.splunk_logs_producer import SplunkLogsProducer
+from data.logs_model.shared import SharedModel
+
+logger = logging.getLogger(__name__)
+
+
+class SplunkLogsModel(SharedModel):
+    """
+    SplunkLogsModel implements model for establishing connection and sending events to Splunk cluster
+    """
+
+    def __init__(self, should_skip_logging=None, producer=None, splunk_config=None):
+        self._should_skip_logging = should_skip_logging
+        self._logs_producer = LogProducerProxy()
+        if producer == "splunk":
+            self._logs_producer.initialize(SplunkLogsProducer(**splunk_config))
+        else:
+            raise Exception("Invalid log producer: %s" % producer)
+
+    def log_action(
+        self,
+        kind_name,
+        namespace_name=None,
+        performer=None,
+        ip=None,
+        metadata=None,
+        repository=None,
+        repository_name=None,
+        timestamp=None,
+        is_free_namespace=False,
+    ):
+
+        if self._should_skip_logging and self._should_skip_logging(
+            kind_name, namespace_name, is_free_namespace
+        ):
+            return
+
+        if repository_name is not None:
+            assert repository is None
+            assert namespace_name is not None
+            repository = model.repository.get_repository(namespace_name, repository_name)
+
+        if timestamp is None:
+            timestamp = datetime.today()
+
+        account_id = None
+        performer_id = None
+        repository_id = None
+
+        if namespace_name is not None:
+            ns_user = model.user.get_namespace_user(namespace_name)
+            if ns_user is not None:
+                account_id = ns_user.id
+
+        if performer is not None:
+            performer_id = performer.id
+
+        if repository is not None:
+            repository_id = repository.id
+
+        kind_id = model.log._get_log_entry_kind(kind_name)
+
+        metadata_json = metadata or {}
+
+        log_data = {
+            "kind": kind_id,
+            "account": account_id,
+            "performer": performer_id,
+            "repository": repository_id,
+            "ip": ip,
+            "metadata_json": metadata_json or {},
+            "datetime": timestamp,
+        }
+
+        try:
+            self._logs_producer.send(json.dumps(log_data, indent=None, sort_keys=True, default=str))
+        except LogSendException as lse:
+            logger.exception("log_action failed", lse)
```

Comment:
Unless you change `SplunkLogsProducer.send` to raise a `LogSendException`, this code will never execute.

Reply:
ah, you're right!